python/CLOUDWATCH_LOG_GROUP_RETENTION_NEVER_EXPIRE_CHECK/CLOUDWATCH_LOG_GROUP_RETENTION_NEVER_EXPIRE_CHECK.py

```diff
@@ -0,0 +1,405 @@
+
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+#####################################
+##           Gherkin               ##
+#####################################
+Rule Name:
+    CLOUDWATCH_LOG_GROUP_RETENTION_NEVER_EXPIRE_CHECK
+Description:
+    Checks whether the AWS CloudWatch Log Group retention period is set to 'Never Expire'. The rule is NON_COMPLIANT if the CloudWatch has a retention period set as 'Never Expire'.
+Trigger:
+    Periodic
+Reports on:
+    AWS::Logs::LogGroup
+Scenarios:
+  Scenario: 1
+    Given: No CloudWatch Log Group is present.
+    Then: Return NOT_APPLICABLE
+  Scenario: 2
+    Given: At least one CloudWatch Log Group is present.
+    And: Log Group has a retention period of 'Never Expire' set (DescribeLogGroups call does not return the 'retentionInDays' name/value pair).
+    Then: Return NON_COMPLIANT
+  Scenario: 3
+    Given: At least one CloudWatch Log Group is present.
+    And: Log Group has a retention period set (DescribeLogGroups call returns the 'retentionInDays' name/value pair).
+    Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::Logs::LogGroup'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    logs_client = get_client('logs', event)
+    evaluations = []
+
+    all_log_groups = get_all_log_groups(logs_client)
+
+    # No log group exists
+    if not all_log_groups:
+        return None
+
+    for each_loggroup in all_log_groups:
+        # NON_COMPLIANT if retentionInDays name/value pair does not exist.
+        if "retentionInDays" not in each_loggroup:
+            evaluations.append(build_evaluation(each_loggroup['logGroupName'], 'NON_COMPLIANT', event, annotation='This CloudWatch Log Group has a retention period set to Never Expire'))
```

Comment:
Can we split the line to multiple line ? If I am not wrong, all customers has some linter in their pipeline which throws error when the Line length is more than a specific number. (Example Flake8 - 79 chars, Pylint is 100 chars)